client/components/Cart.js

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchSingleProduct } from "../store/singleProduct";
+import { addToCart } from "../store/addToCart";
+
+const Cart = () => {
+  const dispatch = useDispatch();
+  //use a react reducer not redux
+  const state = useSelector((state) => state);
+  const [itemCount, setItemCount] = useState(1)
```

Comment:
With how this is set up right now, changing the quantity of any individual item in the cart ends up changing the quantity of all of them. The item quantity in the cart should also be kept in redux, alongside the actual items themselves. The way you have the Orders model on the server to represent a cart should ultimately force this to be fixed anyway when you start updating the server cart